pkg/mtls/secret_manager.go

```diff
@@ -85,7 +85,7 @@ func (mng *secretManager) getOrCreateProvider(cfg *v2.TLSConfig) *sdsProvider {
 		p.config.Store(cfg)
 		v.certificates[certName] = p
 		// set a certificate callback
-		client := GetSdsClient(cfg.SdsConfig.CertificateConfig.Config)
+		client := GetSdsClientV3(cfg.SdsConfig.CertificateConfig.Config)
```

Comment:
Need to choose `GetSdsClientV2` or `GetSdsClientV3` depending on `types.XdsVersion`.